### DISCUSSION TITLE
[Question] How to preselect an arbitrary item?

### DISCUSSION THREAD

**Discussion by T1mL3arn** [john32b/djFlixel#27] [General]
When I create a menu it looks like this:

```
> foo
  bar
  bizz
```

Let say want a selection cursor to be on "bar":

```
  foo
> bar
  bizz
```

How can I do that?

**Comment by john32b** [comment]
There is a function in the `MPage` object called `item_moveCursorTo(itemID)` *(which now that I think about it, it should be named `item_focus(id)`)*

you can get the current active menupage of an flxmenu with `FlxMenu.mpActive`, which returns the currently shown page object, so you need to make sure the page is open first.

so something like
```haxe
var menu = new FlxMenu(...);
menu.createpage(....);
menu.open(); // <--- makes sure mpActive is populated, else it is null
menu.mpActive.item_moveCursorTo('bar'); // <-- focus the item with id `bar`
```

At one point I think I had the option to have a menuPage always focus a custom menuitem once it opens. But I removed it :-/, Also currently I don't have a running haxe toolset to check this, but this should be it I think.

Are you planning on using this when a menu page comes in and it needs to point to another element, or is it changing the focused item in real time?

**Reply by T1mL3arn** [reply]
>when a menu page comes in and it needs to point to another element, 

I guess this. In some subclass I need to slightly change the menu just after its initialization so it should have different menu item selected at the beginning.

**Reply by T1mL3arn** [reply]
Also want to note the need for this pre-selection appeared after seeing that disabled unselectable item was not automatically unset its selection.

**Comment by john32b** [comment]
I'm thinking of making a special markup verb for when creating items, something like 
` -|Options      | link   | @options | autofocus ` ?
the **autofocus** will make that focus whenever the page opens, or something like that
I'll push an update in a couple of days. :cat:

**Comment by john32b** [comment]
Have a look at this : https://github.com/john32b/djFlixel/discussions/28

I chose to use the abbreviated `AF`  since I already used the abbreviated `U` and `D` for unselectable and disabled
